subprojects/gradle/build-verdict/src/test/kotlin/com/avito/android/build_verdict/BuildVerdictPluginConfigurationPhaseTest.kt

```diff
@@ -0,0 +1,129 @@
+            AndroidAppModule(
+                name = appName,
+                buildGradleExtra = """illegal gradle text"""
+            )
+        )
+
+        val result = gradlew(
+            temp,
+            "help",
+            dryRun = true,
+            expectFailure = true
+        )
+        result.assertThat().buildFailed()
+        assertBuildVerdictFileExist(true)
+        assertThat(plainTextBuildVerdict.readText()).isEqualTo(configurationIllegalMethodFails(temp))
+        val actualBuildVerdict = gson.fromJson(jsonBuildVerdict.readText(), BuildVerdict.Configuration::class.java)
+
+        assertThat(actualBuildVerdict.error.message).isEqualTo("Build completed with 2 failures.")
+
+        val errors = (actualBuildVerdict.error as Multi).errors
+
+        assertThat(errors).hasSize(2)
+
+        @Suppress("MaxLineLength")
+        errors[0].assertSingleError(
+            expectedMessageLines = listOf(
+                "$temp/app/build.gradle' line: 9",
+                "A problem occurred evaluating project ':app'."
+            ),
+            expectedCauseMessages = listOf(
+                "A problem occurred evaluating project ':app'.",
+                "Could not find method illegal() for arguments [build 'test-project'] on project ':app' of type org.gradle.api.Project."
+            )
+        )
+
+        errors[1].assertSingleError(
+            expectedMessageLines = listOf(
+                "A problem occurred configuring project ':app'."
+            ),
+            expectedCauseMessages = listOf(
+                "A problem occurred configuring project ':app'.",
+                "compileSdkVersion is not specified. Please add it to build.gradle"
+            )
+        )
+    }
+
+    @Test
+    fun `configuration fails - build verdict contains project dependecy error`() {
+        generateProject(
+            KotlinModule(
+                name = appName,
+                dependencies = setOf(
+                    project(":not-existed")
+                )
+            )
+        )
+
+        val result = gradlew(
+            temp,
+            "help",
+            dryRun = true,
+            expectFailure = true
+        )
+        result.assertThat().buildFailed()
+        assertBuildVerdictFileExist(true)
+        assertThat(plainTextBuildVerdict.readText()).isEqualTo(configurationProjectNotFoundFails(temp))
+
+        val actualBuildVerdict = gson.fromJson(jsonBuildVerdict.readText(), BuildVerdict.Configuration::class.java)
+        val error = actualBuildVerdict.error as Single
+
+        error.assertSingleError(
+            expectedMessageLines = listOf(
+                "$temp/app/build.gradle' line: 9",
+                "A problem occurred evaluating project ':app'."
+            ),
+            expectedCauseMessages = listOf(
+                "A problem occurred evaluating project ':app'.",
+                "Project with path ':not-existed' could not be found in project ':app'."
```

Comment:
I propose to simplify checks and check only the existence of keywords and do not check the full message from Gradle.
Details of formatting are Gradle responsibility and can be changed. I don't want to rely on this because it's not a contract or API.

On this example:
The crucial part is to check that we have `:not-existed` in error message.

Reply:
I will reduce length but not too short because messages become harder to understand